### DISCUSSION TITLE
SFTP connection to primitive ftpd

### DISCUSSION THREAD

**Discussion by dazu89** [doublecmd/doublecmd#1356] [General]
Trying to establish a connection using the pre-bundled WFX FTP plugin to an android device running primitive ftpd, the following error occurs _Error: [6], SSH ERROR Failed opening remote file_ immediately after _Info: [1], CONNECT \Test_ . Unfortunately, I could not find any error description or way to debug this.
Also tried to install and use the new [SFTP plugin for total commander](https://ghisler.ch/board/viewtopic.php?t=19994&sid=2ff757c55aeda9c4dc0c9a8f68817bfb) instead of the old wfx plugin, however, once installed adding a new connection showed in doublecmd with `.txt` extension, and opened a help file not a command, no idea what causes this misbehaviour either.

**Comment by alexx2000** [comment]
Double Commander 1.0.11 and Primitive FTPd 6.16 works fine with my phone. But Primitive FTPd should be [configured correctly](https://github.com/wolpi/prim-ftpd#permission).

**Comment by dazu89** [comment]
Thank you, for your reply, alexx2000. I am using the pre-release 1.1.0 due to availability of a dark theme. Primitive FTPd is not the issue, it works with FileZilla. I tried Primitive FTPd configured with Plain old filesystem, as it allows root access. Android's Storage Access Framework only allows access to non-root folders on my phone. 
Double Commander connects, but then no directories appear in the file-and-folders panel.
```
Info: [1], 
Info: [3], 220 Service ready for new user.
Info: [3], USER Test
Info: [3], 331 User name okay, need password for Test.
Info: [3], PASS ********
Info: [3], 230 User logged in, proceed.
Info: [3], TYPE I
Info: [3], 200 Command TYPE okay.
Info: [3], STRU F
Info: [3], 200 Command STRU okay.
Info: [3], MODE S
Info: [3], 200 Command MODE okay.
Info: [3], REST 0
Info: [3], 350 Restarting at 0. Send STORE or RETRIEVE to initiate transfer.
Info: [3], REST 1
Info: [3], 350 Restarting at 1. Send STORE or RETRIEVE to initiate transfer.
Info: [3], REST 0
Info: [3], 350 Restarting at 0. Send STORE or RETRIEVE to initiate transfer.
Info: [3], FEAT
Info: [3], 211-Extensions supported
Info: [3],  SIZE
Info: [3],  MDTM
Info: [3],  REST STREAM
Info: [3],  LANG en;zh-tw;ja;is
Info: [3],  MLST Size;Modify;Type;Perm
Info: [3],  UTF8
Info: [3],  TVFS
Info: [3],  MD5
Info: [3],  MMD5
Info: [3],  MFMT
Info: [3], 211 End
Info: [3], OPTS UTF8 ON
Info: [3], 200 Command OPTS okay.
Info: [1], CONNECT \Test
Info: [3], PWD
Info: [3], 257 "/" is current directory.
Info: [3], CWD /
Info: [3], 250 Directory changed to /
Info: [3], TYPE A
Info: [3], 200 Command TYPE okay.
Info: [3], PORT 192,168,178,103,249,68
Info: [3], 200 Command PORT okay.
Info: [3], MLSD
Info: [3], 150 File status okay; about to open data connection.
```

**Comment by alexx2000** [comment]
FileZilla works via FTP. DC works fine with my phone via FTP too. Try to enable passive mode in DC connection settings and reconnect.

**Reply by dazu89** [reply]
FileZilla allows both SFTP and FTPS. Following your recommendation, I desisted from using SFTP and tried to configure FTP over TLS instead setting the _Use passive mode for transfers (like a WWW browser)_ option in the ‹Advanced› tab – success! Thank you for your insight and time, I will read up on the passive mode.

**Comment by alexx2000** [comment]
Important information about FTP over TLS #437
